question-servers/freeform.js

```diff
@@ -1854,7 +1854,7 @@ module.exports = {
   async getCacheKey(course, data, context) {
     try {
       const commitHash = await courseUtil.getOrUpdateCourseCommitHashAsync(course);
-      const dataHash = hash('sha1').update(JSON.stringify({ data, context })).digest('base64');
+      const dataHash = objectHash({ data, context }, { algorithm: 'sha1', encoding: 'base64' });
```

Comment:
Is this still running on the main thread? You mentioned offloading computation via the web crypto API?

Reply:
It is indeed still running on the main thread; for better or worse `object-hash` doesn't utilize the `crypto.subtle` API. TBH I think the fact that the `createHash` API supports incremental `.update()` calls with new data is pretty critical to the way that `object-hash` works.

I think that this should virtually always be fast enough on the main thread to not notice the fact that this is synchronous.

Reply:
Ok, sounds good, the speedup is a huge win in any case.